libass/ass_blur.c

```diff
@@ -849,8 +849,14 @@ bool ass_gaussian_blur(const BitmapEngine *engine, Bitmap *bm, double r2)
     int end_w = ((w + offset) & ~((1 << blur.level) - 1)) - 4;
     int end_h = ((h + offset) & ~((1 << blur.level) - 1)) - 4;
 
+    if (end_w < 1 || end_h < 1 || end_w >= INT_MAX / 4)
```

Comment:
`< 1` check doesn't have sense here. It's impossible to produce negative values (and zero) in regular case. The only problem here is overflow in `w + offset` and `h + offset`, but it's UB and should be checked before operation.

Reply:
I did it because of the ``(...) - 4`` above. Are you sure the left side can't be lower than 5?

Reply:
Yes. It can be seen that `offset >= (12 << blur.level)` so `end_w > 8`. It's exact calculation of final size of bitmap after bunch of different stages, but each stage have positive minimal result size.